internal/controller/dependencyupdatecheck_controller.go

```diff
@@ -18,55 +18,216 @@ package controller
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(secretList.Items) == 0 {
+		// No secrets to merge
+		return nil, nil
+	}
+
+	logger.Info(fmt.Sprintf("Found %d secrets to merge", len(secretList.Items)))
+
+	mergedAuths := make(map[string]interface{})
+	for _, secret := range secretList.Items {
+		if secret.Type == corev1.SecretTypeDockerConfigJson {
+			data, exists := secret.Data[".dockerconfigjson"]
+			if !exists {
+				// No .dockerconfigjson section
+				logger.Info("Found secret without .dockerconfigjson section")
+				return nil, nil
+			}
+
+			var dockerConfig map[string]interface{}
+			if err := json.Unmarshal(data, &dockerConfig); err != nil {
+				return nil, err
+			}
+
+			auths, exists := dockerConfig["auths"].(map[string]interface{})
+			if !exists {
+				continue
+			}
+
+			for registry, creds := range auths {
+				mergedAuths[registry] = creds
+			}
+		}
+	}
+
+	mergedDockerConfig := map[string]interface{}{
+		"auths": mergedAuths,
+	}
+
+	if len(mergedAuths) == 0 {
+		logger.Info("Merged auths empty, skipping creation of secret")
+
+		return nil, nil
+	}
+
+	mergedConfigJson, err := json.Marshal(mergedDockerConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	timestamp := time.Now().Unix()
+	name := fmt.Sprintf("renovate-image-pull-secrets-%d-%s", timestamp, RandomString(5))
+
+	newSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: MintMakerNamespaceName,
+		},
+		Type: corev1.SecretTypeDockerConfigJson,
+		Data: map[string][]byte{
+			".dockerconfigjson": []byte(mergedConfigJson),
 		},
-		eventRecorder:  eventRecorder,
-		jobCoordinator: renovate.NewJobCoordinator(client, scheme),
 	}
+
+	if err := r.Client.Create(ctx, newSecret); err != nil {
+		return nil, err
+	}
+
+	return newSecret, nil
 }
 
-//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=dependencyupdatechecks,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=dependencyupdatechecks/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=dependencyupdatechecks/finalizers,verbs=update
+// createPipelineRun creates and returns a new PipelineRun
+func (r *DependencyUpdateCheckReconciler) createPipelineRun(ctx context.Context, resources *loader.ProcessingResources) (*tektonv1beta1.PipelineRun, error) {
+
+	logger := log.FromContext(ctx)
+	timestamp := time.Now().Unix()
+	name := fmt.Sprintf("renovate-pipelinerun-%d-%s", timestamp, RandomString(5))
+
+	renovateImageUrl := os.Getenv(RenovateImageEnvName)
+	if renovateImageUrl == "" {
+		renovateImageUrl = DefaultRenovateImageUrl
+	}
+
+	// Create a merged secret for private registries
+	registry_secret, err := r.createMergedPullSecret(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var renovateCmd []string
+	// TODO: this needs to be changed to define the RENOVATE_TOKEN once we will get the secret for the component
+	for _, task := range tasks {
```

Comment:
`tasks` is not defined. To keep things simple at this stage, I suggest we avoid moving the code from `renovate/job.go` to the new controller for now. Instead, focus on making the new controller(s) create a simple `PipelineRun` that only prints a message, as you're currently doing.

There are other undefined variables in this PR as well.

After preparing the new controller(s), you can try to run it locally with (if you have a local `minikube` cluster):
```
$ kubectl apply -k https://github.com/konflux-ci/application-api/config/crd

$ kubectl apply -k https://github.com/redhat-appstudio/infra-deployments/components/mintmaker/development

# then let's remove the old controller by removing the deployment
$ kubectl delete deployment/mintmaker-controller-manager

# now we can run the new controller(s) by manual
$ go run cmd/main.go
```

Reply:
Thank you for the review. I agree, I also tried to run this locally, but still it doesn't run, I still have some errors and tests aren't defined. That's how far I've got on Friday. I'll follow your suggestions and perhaps we can merge it, and then I'll have another PR to continue with this work. But at least if other folks wants to contribute they aren't blocked anymore.